Orange/widgets/visualize/owboxplot.py

```diff
@@ -972,6 +1027,50 @@ def send_report(self):
         if text:
             self.report_caption(text)
 
+    class Label(QGraphicsSimpleTextItem):
+        """Boxplot Label with settable maxSize"""
+        # Minimum width to display label text
+        MIN_LABEL_WIDTH = 25
+
+        # amount of whitespace between the cut point on the label and
+        # the right bar boundary
+        RIGHT_PADDING = 3
+
+        __max_width = None
+
+        def maxWidth(self):
+            return self.__max_width
+
+        def setMaxWidth(self, max_width):
+            self.__max_width = max_width
+
+        def paint(self, painter, option, widget):
+            """Overrides QGraphicsSimpleTextItem.paint
+
+            If text is wider than maxWidth, a white rect is drawn to obscure
+            letters that extend beyond allowed width.
+            """
+            super().paint(painter, option, widget)
+
+            if self.__max_width is None or self.boundingRect().width() < self.__max_width:
+                return
+
```

Comment:
@ales-erjavec Do you perhaps have any ideas how to limit label painting to some rect? So that we would not need to draw white rects over label parts?

Reply:
* set clip rect on the painter
* actually implement bounded text painting (QPainter has quite a few drawText overloads for bounded text painting, also see QFontMetrics.elidedText)